cmplibrary/src/test/java/com/sourcepoint/cmplibrary/util/extensions/CcpaCSExtKtTest.kt

```diff
@@ -0,0 +1,185 @@
+package com.sourcepoint.cmplibrary.util.extensions
+
+import com.sourcepoint.cmplibrary.assertEquals
+import com.sourcepoint.cmplibrary.data.network.model.optimized.CcpaCS
+import com.sourcepoint.cmplibrary.model.exposed.CcpaStatus
+import org.junit.Test
+
+internal class CcpaCSExtKtTest {
+
+    /**
+     * Test case that covers the case when:
+     * applies == null
+     * then the consent string should be == "1---"
+     */
+    @Test
+    fun `generateConsentString - WHEN called with applies as null THEN should return default CCPA consent string`() {
+        // GIVEN
+        val ccpaCS = CcpaCS(
```

Comment:
Creating a `CcpaCS` seems a bit cumbersome. Since the only thing that we care (in this test) is changing the values of `signedLspa`, `applies` and the consent status, you could create a helper method like this:
```kotlin
fun genCCPA(signedLspa: Boolean, applies: Boolean, status: CCPAStatus): String {
    return CcpaCS(
            actions = null,
            applies = applies,
            ccpaApplies = null,
            consentedAll = null,
            cookies = null,
            dateCreated = null,
            gpcEnabled = null,
            newUser = null,
            rejectedAll = null,
            rejectedCategories = null,
            rejectedVendors = null,
            signedLspa = signedLspa,
            status = status,
            uspstring = "",
            uuid = null,
        )
}
```

And use the method on each test case.

Reply:
I changed the place for `generateCcpaUspString()` method, so I changed the place for test suite as well, created this helper method you've mentioned, so now from my pov it looks clean